R/R-duckdb/Portfile

```diff
@@ -22,10 +26,28 @@ compiler.cxx_standard 2017
 # https://github.com/duckdb/duckdb-r/issues/152
 patchfiles-append   patch-fix-cmath.diff
 
-# https://github.com/duckdb/duckdb/issues/7009
-if {[string match macports-gcc* ${configure.compiler}] && (${configure.build_arch} in [list i386 ppc])} {
-    patchfiles-append \
-                    patch-libatomic.diff
+if {${os.platform} eq "darwin" && ${os.major} < 16} {
+    patchfiles-append   patch-Makevars.diff
+
+    post-patch {
+        # https://github.com/duckdb/duckdb/issues/7009
+        if {[string match macports-gcc* ${configure.compiler}] && (${configure.build_arch} in [list i386 ppc])} {
+            reinplace "s|@ATOMIC@|-latomic|" ${worksrcpath}/src/Makevars
+        } else {
+            reinplace "s|@ATOMIC@||" ${worksrcpath}/src/Makevars
+        }
+
+        # https://trac.macports.org/ticket/69902
+        # On 10.6 and earlier LegacySupport is already used,
```

Comment:
What causes legacysupport to be used on 10.6 and earlier but not later? I'm looking specifically at the line `legacysupport.newest_darwin_requires_legacy 15` above.

Reply:
@ryandesign `R` portfile sets `legacysupport.newest_darwin_requires_legacy 10`, so for 10.6 and below flags are baked into `R`, it seems. On my system there are used for every port automatically, and apparently on buildbots it is the same.

Reply:
I guess we do not want to enable it via `R` portfile for everything just to fix a single port. So I mark it as resolved.